.github/workflows/scripts/start-cil-runner.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+# This script is used to start a CNCF CIL runner 
+
+token=$1
+hostname=$2
+
+# Generate random number from datastamp as the hostname of runner
+label=$(date +%N)
+
+hostname="$hostname-$label"
+echo "Creating CNCF CIL machine: $hostname..."
+
+# Use user_data_scripts to register the CNCF CIL runner as a self-hosted runner
+user_data_scripts="#cloud-config\nusers:\n    - default\n    - name: smp\n      groups: sudo, docker\n      sudo: ALL=(ALL) NOPASSWD:ALL\n      lock_passwd: true\nruncmd:\n    - [runuser, -l, smp, -c, \'mkdir actions-runner && cd actions-runner\']\n    - [runuser, -l, smp, -c, \'curl -o actions-runner-linux-x64-2.287.1.tar.gz -L https://github.com/actions/runner/releases/download/v2.287.1/actions-runner-linux-x64-2.287.1.tar.gz\']\n    - [runuser, -l, smp, -c, \'tar xzf ./actions-runner-linux-x64-2.287.1.tar.gz\']\n    - [runuser, -l, smp, -c, \'export RUNNER_ALLOW_RUNASROOT=1\']\n    - [runuser, -l, smp, -c, \'./config.sh --url https://github.com/$REPOSITORY --token $REG_TOKEN --labels $hostname >> github-action-registeration.log\']\n    - [runuser, -l, smp, -c, \'./run.sh >> github-action-registeration.log\']"
+
+# TODO: the options "operating_system", "facility", "plan" are hardcoded now, we should make them configurable
+# https://metal.equinix.com/developers/api/devices/#devices-createdevice
+device_id=$(curl -X POST -H "X-Auth-Token: $token" -s -H "Content-Type: application/json" \
+-d '{"operating_system": "ubuntu_20_04", "facility": "da11", "plan": "c3.small.x86", "hostname": "'"${hostname}"'", "userdata": "'"${user_data_scripts}"'"}' \
+https://api.equinix.com/metal/v1/projects/96a9d336-541b-42f7-9827-d845010da550/devices | jq -r .id)
+if [[ -z $device_id ]]; then
+    echo "ERROR: Failed to create CNCF CIL machine: $hostname..."
+    exit 1
+fi
+
+# Wait 10 minutes until the machine is running 
```

Comment:
Any reason to wait 10 minutes? Is standard waiting time for an instance to come up mentioned somewhere?

Reply:
There is no standard waiting time in my opinion. My thought about waiting 10 minutes is that if it takes too long to wait for machine to be running, there must be something wrong with it. 
However, on my second thought, it would be better if we could take more advantage of "state" field instead of just waiting 10 minutes:
1. If "state" == "provisioning", sleep 10s...
2. If "state" == "active", echo "Machine successfully created!" and continue.
3. If "state" == "failed", echo "Failed to create machine" and exit.
How do you think about this?

Reply:
@gyohuangxin this new flow makes a lot of sense. But let's capture this in another issue. There might be slight experimentation required here.

Reply:
I understand this issue and will be creating an issue ticket for this one

Reply:
Thanks, please go ahead.